File: 1380-lucky-numbers-in-a-matrix/1380-lucky-numbers-in-a-matrix.cpp
class Solution {
public:
    void maxmin(vector<vector<int>>& matrix,vector<int>& v,int r, int c,int m,int n){
        int min = INT_MAX;
        int max = -1;
        int k=-1;
        int l=-1;
        int a = -1;
        int b = -1;
        for(int i=0;i<n;i++){
            if(matrix[r][i]<min){
                min = matrix[r][i];
                k=r;
                l=i;
            }
        }
        for(int i=0;i<m;i++){
            if(matrix[i][c]>max){
                max = matrix[i][c];
                a=i;
                b=c;
            }
        }
        if(max==min && (k==a && l==b)) v.push_back(max);
    }
    vector<int> luckyNumbers(vector<vector<int>>& matrix) {
        int m = matrix.size();
        int n = matrix[0].size();
        vector<int> v;
        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){
                maxmin(matrix,v,i,j,m,n);
              }
        }
        return v;
    }
};

File: 1380-lucky-numbers-in-a-matrix/1380-lucky-numbers-in-a-matrix.java
class Solution {
    public List<Integer> luckyNumbers(int[][] matrix) {
        int m = matrix.length;
        int n = matrix[0].length;
        int c=-1;
        List<Integer> ans = new ArrayList<>();
        for(int i=0;i<m;i++){
            int minm = Integer.MAX_VALUE;
            for(int j=0;j<n;j++){
                if(minm>matrix[i][j]){
                    minm = matrix[i][j];
                    c = j;
                    }
            }
            int maxm = -1;
            for(int k=0;k<m;k++){
                    if(matrix[k][c]>maxm) maxm = matrix[k][c];
                }
            if(maxm==minm){
                ans.add(maxm);
                break;
            }
        }
        return ans;
    }
}